enumerables.rb

```diff
@@ -0,0 +1,67 @@
+module Enumerable
+  def my_each
+    return to_enum unless block_given?
+
+    length.times { |i| yield(self[i]) }
+    self
+  end
+
+  def my_each_with_index
+    return to_enum unless block_given?
+
+    length.times { |i| yield(self[i], i) }
+    self
+  end
+
+  def my_select
+    return to_enum unless block_given?
+
+    array = []
+    my_each { |i| array << i if yield(i) }
+    array
+  end
+
+  def my_all?
+    output = true
+    my_each { |i| break output = false unless yield(i) }
+    output
+  end
+
+  def my_any?
+    output = false
+    my_each { |i| break output = true unless yield(i) }
+    output
+  end
```

Comment:
- returns true if the block ever returns a value other than false or nil
- returns false if the block ever returns a value false or nil
- when no block or argument is given returns true if at least one of the collection is not false or nil
- when a class is passed as an argument returns true if at least one of the collection is a member of such class
- when a Regex is passed as an argument returns false if none of the collection matches the Regex
- when a pattern other than Regex or a Class is given returns false if none of the collection matches the pattern